src/types.d.ts

```diff
@@ -1,18 +1,5 @@
-export interface EventEmitterFactory {
-  new(): EventEmitter;
-}
+export type SelectFn = function (Uint8Array, Uint8Array[]): number
+export type ValidateFn = function (Uint8Array, Uint8Array): Promise<void>
```

Comment:
If you add a typedef to `/src/index.js` like:

```js
/**
 * @typedef {import('./types').SelectFn} SelectFn
 * @typedef {import('./types').ValidateFn} ValidateFn
 * @typedef {import('./types').DhtSelectors} DhtSelectors
 * @typedef {import('./types').DhtValidators} DhtValidators
 */
```

then you can use them from other modules like:

```js
/**
 * @typedef {import('libp2p-interfaces').SelectFn} SelectFn
 */
```

otherwise you have to do:

```js
/**
 * @typedef {import('libp2p-interfaces/src/types').SelectFn} SelectFn
 */
```

Which is ok, but needs the `typesVersions` hack and prevents you from moving that file in future.

Reply:
This was causing:

```
src/validator.js:15:19 - error TS2306: File '/Users/vsantos/work/pl/gh/libp2p/js-libp2p-record/node_modules/libp2p-interfaces/dist/src/index.d.ts' is not a module.
15  * @param {import('libp2p-interfaces').DhtValidators} validators
```

After generated

```
type SelectFn = import('./types').SelectFn;
type ValidateFn = import('./types').ValidateFn;
type DhtSelectors = import('./types').DhtSelectors;
type DhtValidators = import('./types').DhtValidators;
//# sourceMappingURL=index.d.ts.map
```

I could not easily work around this issue, so I reverted to the original solution for now